docs/en/index.md

```diff
@@ -159,33 +159,190 @@ Nested GridFields
-$grid->getConfig()->addComponent(GridFieldNestedForm::create());
+class ParentNode extends DataObject
+{
+    ...
+
+    private static $has_many = [
+        'ChildNodes' => BranchNode::class,
+    ];
+}
 
-// Usage with custom relation
-$grid->getConfig()->addComponent(GridFieldNestedForm::create()->setRelationName('MyRelation'));
 ```
 
-You can define your own custom GridField config for the nested GridField configuration by implementing a `getNestedConfig`
-on your nested model (should return a `GridField_Config` object).
 ```php
-class NestedObject extends DataObject
+class ChildNode extends DataObject
 {
-	private static $has_one = [
-		'Parent' => ParentObject::class
+    ...
+
+    private static $has_one = [
+        'ParentNode' => ParentNode::class,
+    ];
+
+    private static $has_many = [
+        'GrandChildNodes' => GrandChildNode::class,
+    ];
+
+    public function getNestedConfig(): GridFieldConfig
+    {
+        $config = new GridFieldConfig_RecordEditor();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('GrandChildNodes'));
+        return $config;
+    }
+}
+
+```
+
+```php
+class GrandChildNode extends DataObject
+{
+    ...
+
+    private static $has_one = [
+        'ChildNode' => ChildNode::class,
+    ];
+}
+
+```
+
+```php
+class MyAdminSection extends ModelAdmin
+{
+    private static string $url_segment = 'my-admin-section';
+    private static string $menu_title = 'My Admin Section';
+
+    private static array $managed_models = [
+		ParentNode::class
 	];
 
-	public function getNestedConfig(): GridFieldConfig
-	{
-		$config = new GridFieldConfig_RecordViewer();
-		return $config;
-	}
+    protected function getGridFieldConfig(): GridFieldConfig
+    {
+        $config = parent::getGridFieldConfig();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('ChildNodes'));
+
+        return $config;
+    }
 }
+
 ```
 
-You can also modify the default config (a `GridFieldConfig_RecordEditor`) via an extension to the nested model class, by implementing
-`updateNestedConfig`, which will get the config object as the first parameter.
+There is also the possibility to use Nested GridField with the data structure ArrayList. To do this, you can use the following approach.
+
+```php
+class MyDataSet extends DataObject
+{
+    ...
+
+    public function getMyArrayList() {
+        $list = ArrayList::create();
+        $data = MyDataObject::get();
+
+        foreach ($data as $value) {
+            $list->push($value);
+        }
+
+        return $list;
+    }
+}
+
+```
+
+```php
+class MyAdminSection extends ModelAdmin
+{
+    ...
+
+    private static array $managed_models = [
+		MyDataSet::class
+	];
+
+    protected function getGridFieldConfig(): GridFieldConfig
+    {
+        $config = parent::getGridFieldConfig();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('getMyArrayList'));
+
+        return $config;
+    }
+}
```

Comment:
Probably this test case should be updated with explanation how to set relation name, if user wants to show nested ArrayList in another ArrayList.

Reply:
Yes please - even just a quick sentence saying something like this would be good:
> Notice that instead of the name of a relation, we're passing a method name into `getMyArrayList()`. That method must return an instance of `SS_List` to be used in the `GridField`.

Reply:
Updated